src/server.rs

```diff
@@ -43,39 +44,29 @@ impl Default for ServerPlugin {
 
 impl Plugin for ServerPlugin {
     fn build(&self, app: &mut App) {
-        app.add_plugins((
-            RenetServerPlugin,
-            NetcodeServerPlugin,
-            RemovalTrackerPlugin,
-            DespawnTrackerPlugin,
-        ))
-        .init_resource::<AckedTicks>()
-        .init_resource::<TicksMap>()
-        .init_resource::<MinRepliconTick>()
-        .init_resource::<ClientEntityMap>()
-        .configure_sets(PreUpdate, ServerSet::Receive.after(RenetReceive))
-        .configure_sets(
-            PostUpdate,
-            ServerSet::Send
-                .before(RenetSend)
-                .run_if(resource_changed::<RepliconTick>()),
-        )
-        .add_systems(
-            PreUpdate,
-            (Self::acks_receiving_system, Self::acks_cleanup_system)
-                .in_set(ServerSet::Receive)
-                .run_if(resource_exists::<RenetServer>()),
-        )
-        .add_systems(
-            PostUpdate,
-            (
-                Self::replication_sending_system
-                    .map(Result::unwrap)
-                    .in_set(ServerSet::Send)
+        app.add_plugins((RenetServerPlugin, NetcodeServerPlugin))
+            .init_resource::<ClientsInfo>()
+            .init_resource::<LastChangeTick>()
+            .init_resource::<ClientEntityMap>()
+            .configure_sets(PreUpdate, ServerSet::Receive.after(RenetReceive))
+            .configure_sets(PostUpdate, ServerSet::Send.before(RenetSend))
+            .add_systems(
+                PreUpdate,
+                (Self::acks_receiving_system, Self::disconnect_cleanup_system)
+                    .in_set(ServerSet::Receive)
                     .run_if(resource_exists::<RenetServer>()),
-                Self::reset_system.run_if(resource_removed::<RenetServer>()),
-            ),
-        );
+            )
+            .add_systems(
+                PostUpdate,
+                (
+                    Self::replication_sending_system
+                        .map(Result::unwrap)
```

Comment:
I agree, our politic is to never panic on server. But the panic in this system can happen only on serialization (programmer's error).

Reply:
I think it's impossible to introduce a client attack vector here. We only iterate over the world, serialize data and send messages.

Reply:
> It's easy enough, just write a little helper function:

But this way it will early-return. I.e. in the middle of preparing packets it will print error and exit without even sending the data.

> Users can define their own arbitrary serialization functions, which can easily expose an attack vector.

Like serializing a component based on input from clients? A bit hard to imagine something like this, but something like this should be validated in any way...

You can't even have I/O errors here. Only if something goes wrong with serialization logic and it will panic on first such component.